lib/bloc/blocs/user_bloc.dart

```diff
@@ -0,0 +1,58 @@
+
+import 'package:bloc/bloc.dart';
+import 'package:injectable/injectable.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:structure_flutter_mobile/bloc/events/user_event.dart';
+import 'package:structure_flutter_mobile/bloc/states/user_state.dart';
+import 'package:structure_flutter_mobile/data/model/user.dart';
+import 'package:structure_flutter_mobile/di/injection.dart';
+import 'package:structure_flutter_mobile/repositories/user_repository.dart';
+
+@singleton
+class UserGitBloc extends Bloc<UserGitEvent, UserGitState> {
+  final UserRepository _userRepository = getIt<UserRepository>();
+
+  UserGitBloc();
+
+  @override
+  Stream<UserGitState> transform(Stream<UserGitEvent> events,
+      Stream<UserGitState> Function(UserGitEvent event) next) {
+    return super.transform(
+        (events as Observable<UserGitEvent>)
+            .debounceTime(Duration(milliseconds: 500)),
+        next);
+  }
+
+  @override
+  UserGitState get initialState => UserGitUnInitialized();
+
+  @override
+  Stream<UserGitState> mapEventToState(UserGitEvent event) async* {
+    if (event is Fetch && !_hasReachedMax(currentState)) {
+      try {
+        if (currentState is UserGitUnInitialized) {
+          final users = await fetchUserGit(1);
+          yield UserGitLoaded(users: users, hasReachMax: false);
+        } else if (currentState is UserGitLoaded) {
+          var page =
+              ((currentState as UserGitLoaded).users.length / 10).ceil() + 1;
+          final users = await fetchUserGit(page);
+          yield users.length < 10
+              ? (currentState as UserGitLoaded).copyWith(hasReachedMax: true)
+              : UserGitLoaded(
```

Comment:
I think we should save variable '_page', initial value is 1 and plus one after each fetch data

Reply:
Actually we dont need count when we fetch data. make it easy from data we have. it's flexible and beside we have a variable 'isReachMax' so we never load wrong data. BTW this is just demo.

Reply:
Yep, I totally agree with you!